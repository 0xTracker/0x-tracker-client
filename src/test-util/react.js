import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import React from 'react';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

export { renderWithRouter };
