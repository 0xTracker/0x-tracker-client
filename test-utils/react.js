import { createMemoryHistory } from 'history';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { Router } from 'react-router-dom';
import React from 'react';

import reducer from '../src/redux/reducer';

function renderWithRouter(
  element,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {},
) {
  return {
    ...render(<Router history={history}>{element}</Router>),
    history,
  };
}

const renderWithAppContext = (
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
    initialState,
    store = createStore(reducer, initialState),
  } = {},
) => ({
  ...render(
    <Provider store={store}>
      <Router history={history}>{component}</Router>
    </Provider>,
  ),
  history,
});

export { renderWithAppContext, renderWithRouter };
