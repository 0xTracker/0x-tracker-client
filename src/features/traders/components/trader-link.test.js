import React from 'react';

import { renderWithRouter } from '../../../test-util/react';
import TraderLink from './trader-link';

it('should render a trader link', () => {
  const { container } = renderWithRouter(
    <TraderLink address="0xb14232b0204b2f7bb6ba5aff59ef36030f7fe38b">
      Hello World
    </TraderLink>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
