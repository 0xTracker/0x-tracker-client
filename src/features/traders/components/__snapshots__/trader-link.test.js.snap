// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a trader link 1`] = `
<a
  class="link__InternalLink-sc-1m1n17q-1 IiDhz"
  href="/traders/0xb14232b0204b2f7bb6ba5aff59ef36030f7fe38b"
>
  Hello World
</a>
`;
